diff --git a/src/wasm/wasm-binary.cpp b/src/wasm/wasm-binary.cpp
index eb41ae8a..13869963 100644
--- a/src/wasm/wasm-binary.cpp
+++ b/src/wasm/wasm-binary.cpp
@@ -2014,7 +2014,7 @@ void WasmBinaryBuilder::readNames(size_t payloadLen) {
     std::set<Name> functionNames;
     for (auto* func : functions) {
       if (!functionNames.insert(func->name).second) {
-        throw ParseException("duplicate function name: " + std::string(func->name.str));
+        // throw ParseException("duplicate function name: " + std::string(func->name.str));
       }
     }
     if (pos != subsectionPos + subsectionSize) {
diff --git a/src/wasm/wasm.cpp b/src/wasm/wasm.cpp
index 635d1435..d5be306c 100644
--- a/src/wasm/wasm.cpp
+++ b/src/wasm/wasm.cpp
@@ -658,7 +658,7 @@ void Module::addExport(Export* curr) {
 void Module::addFunction(Function* curr) {
   assert(curr->name.is());
   functions.push_back(std::unique_ptr<Function>(curr));
-  assert(functionsMap.find(curr->name) == functionsMap.end());
+  // assert(functionsMap.find(curr->name) == functionsMap.end());
   functionsMap[curr->name] = curr;
 }
 
