diff -ruN libiconv-1.16/build-aux/config.sub libiconv-1.16-patched/build-aux/config.sub
--- libiconv-1.16/build-aux/config.sub	2019-04-27 03:29:05.000000000 +0900
+++ libiconv-1.16-patched/build-aux/config.sub	2019-11-23 21:36:56.794790501 +0900
@@ -135,7 +135,7 @@
 			| linux-newlib* | linux-musl* | linux-uclibc* | uclinux-uclibc* \
 			| uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* \
 			| netbsd*-eabi* | kopensolaris*-gnu* | cloudabi*-eabi* \
-			| storm-chaos* | os2-emx* | rtmk-nova*)
+			| storm-chaos* | os2-emx* | rtmk-nova* | unknown-wasm)
 				basic_machine=$field1
 				os=$maybe_os
 				;;
@@ -1511,6 +1511,8 @@
 		;;
 	ios)
 		;;
+	unknown-wasm)
+		;;
 	none)
 		;;
 	*-eabi)
diff -ruN libiconv-1.16/lib/loop_wchar.h libiconv-1.16-patched/lib/loop_wchar.h
--- libiconv-1.16/lib/loop_wchar.h	2018-09-18 01:06:14.000000000 +0900
+++ libiconv-1.16-patched/lib/loop_wchar.h	2019-11-23 21:40:43.599165612 +0900
@@ -36,7 +36,9 @@
 # include <wchar.h>
 # define BUF_SIZE 64  /* assume MB_LEN_MAX <= 64 */
   /* Some systems, like BeOS, have multibyte encodings but lack mbstate_t.  */
+#ifndef __wasm__
   extern size_t mbrtowc ();
+#endif
 # ifdef mbstate_t
 #  define mbrtowc(pwc, s, n, ps) (mbrtowc)(pwc, s, n, 0)
 #  define mbsinit(ps) 1
diff -ruN libiconv-1.16/libcharset/build-aux/config.sub libiconv-1.16-patched/libcharset/build-aux/config.sub
--- libiconv-1.16/libcharset/build-aux/config.sub	2019-04-27 03:29:05.000000000 +0900
+++ libiconv-1.16-patched/libcharset/build-aux/config.sub	2019-11-23 21:39:26.892696874 +0900
@@ -135,7 +135,7 @@
 			| linux-newlib* | linux-musl* | linux-uclibc* | uclinux-uclibc* \
 			| uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* \
 			| netbsd*-eabi* | kopensolaris*-gnu* | cloudabi*-eabi* \
-			| storm-chaos* | os2-emx* | rtmk-nova*)
+			| storm-chaos* | os2-emx* | rtmk-nova* | unknown-wasm)
 				basic_machine=$field1
 				os=$maybe_os
 				;;
@@ -1511,6 +1511,8 @@
 		;;
 	ios)
 		;;
+	unknown-wasm)
+		;;
 	none)
 		;;
 	*-eabi)
