diff --git a/include/llvm/CodeGen/TargetLowering.h b/include/llvm/CodeGen/TargetLowering.h
index f9a61b8bf1a..b8213a09085 100644
--- a/include/llvm/CodeGen/TargetLowering.h
+++ b/include/llvm/CodeGen/TargetLowering.h
@@ -2156,7 +2156,7 @@ public:
   /// Return true if an fabs operation is free to the point where it is never
   /// worthwhile to replace it with a bitwise operation.
   virtual bool isFAbsFree(EVT VT) const {
-    assert(VT.isFloatingPoint());
+    // assert(VT.isFloatingPoint());
     return false;
   }
 
